File: WIN.TECHNICAL.VB.UTILS/Security/Permission.vb
Imports WIN.SECURITY.Core

Public Class Permission
   Inherits AbstractPersistenceObject
   Implements IPermission

   Private _profile As IProfile
   Private _fullName As String = ""
   Private _alias As String = ""
   Private _macro As String = ""

   Public Property FullMethodName() As String Implements IPermission.FullMethodName
      Get
         Return _fullName
      End Get
      Set(ByVal value As String)
         _fullName = value
      End Set
   End Property

   Public Property IDPermission() As Integer Implements IPermission.ID
      Get
         If MyBase.Key IsNot Nothing Then
            Return MyBase.Key.Value(0)
         End If
         Return -1
      End Get
      Set(ByVal value As Integer)
         Throw New NotImplementedException
      End Set
   End Property

   

   Public Property [Alias]() As String Implements SECURITY.Core.IPermission.Alias
      Get
         Return _alias
      End Get
      Set(ByVal value As String)
         _alias = value
      End Set
   End Property

   Public Property Macroarea() As String Implements SECURITY.Core.IPermission.Macroarea
      Get
         Return (_macro)
      End Get
      Set(ByVal value As String)
         _macro = value
      End Set
   End Property

   Public Property Profile() As SECURITY.Core.IProfile Implements SECURITY.Core.IPermission.Profile
      Get
         Return _profile
      End Get
      Set(ByVal value As SECURITY.Core.IProfile)
         _profile = value
      End Set
   End Property
End Class

File: WIN.TECHNICAL.SECURITY/Core/Permission.cs
using System;
using System.Collections.Generic;
using System.Text;

using WIN.SECURITY.Core;

namespace WIN.SECURITY.Core
{
    [Serializable]
    public class Permission : IPermission
    {
        private int _id;
        private IProfile _profile;
        private string _fullMethodName;
        private string _macroarea;
        private string _alias;

        public int ID
        {
            get { return _id; }
            set { _id = value; }
        }

        public IProfile Profile
        {
            get { return _profile; }
            set { _profile = value; }
        }

        public string FullMethodName
        {
            get { return _fullMethodName; }
            set { _fullMethodName = value; }
        }

        public string Macroarea
        {
            get { return _macroarea; }
            set { _macroarea = value; }
        }

        public string Alias
        {
            get { return _alias; }
            set { _alias = value; }
        }
    }
}
